File: app/src/main/java/net/feheren_fekete/applist/applistpage/viewmodel/AppItem.java
package net.feheren_fekete.applist.applistpage.viewmodel;


public class AppItem extends StartableItem {
    private String mPackageName;
    private String mClassName;
    private String mAppName;

    public AppItem(long id,
                   String packageName,
                   String className,
                   String appName) {
        super(id);
        mPackageName = packageName;
        mClassName = className;
        mAppName = appName;
    }

    public String getPackageName() {
        return mPackageName;
    }

    public String getClassName() {
        return mClassName;
    }

    @Override
    public String getName() {
        return mAppName;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || !(o instanceof AppItem)) {
            return false;
        }
        AppItem other = (AppItem) o;
        return mPackageName.equals(other.getPackageName())
                && mClassName.equals(other.getClassName());
    }
}

File: app/src/main/java/net/feheren_fekete/applist/applistpage/viewmodel/AppItem.kt
package net.feheren_fekete.applist.applistpage.viewmodel


class AppItem(id: Long,
              val packageName: String,
              val className: String,
              val versionCode: Long,
              name: String,
              customName: String,
              customIconPath: String,
              parentSectionId: Long) : StartableItem(id, name, customName, customIconPath, parentSectionId) {

    override fun equals(other: Any?): Boolean {
        if (other == null || other !is AppItem) {
            return false
        }
        return id == other.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

}
